Disables image preloading when site creation is disabled

antonis opened the pull request:
Partially #17154

## Description
This PR disables site creation thumbnail preloading when the site creation is disabled.

Checking the [stack trace of the linked issue](https://a8c.sentry.io/share/issue/556f598b913648d39b199ab5c3188bae/
) thumbnail [preloading is triggered on the WordPress app](https://github.com/wordpress-mobile/WordPress-Android/blob/402eb0ae49f3df53741154b41727f8d292975462/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVM.kt#L185) even when the site creation functionality is disabled.
```
java.util.concurrent.ExecutionException: com.bumptech.glide.load.engine.GlideException: Failed to load resource
There was 1 root cause:
com.android.volley.NoConnectionError(java.net.ConnectException: Failed to connect to s0.wp.com/192.0.77.32:443)
 call GlideException#logRootCauses(String) for more detail
    at com.bumptech.glide.request.RequestFutureTarget.doGet(RequestFutureTarget.java:217)
    at com.bumptech.glide.request.RequestFutureTarget.get(RequestFutureTarget.java:130)
    at org.wordpress.android.util.image.ImageManager.preload(ImageManager.kt:315)
    at org.wordpress.android.ui.sitecreation.SiteCreationMainVM$preloadThumbnails$1.invokeSuspend(SiteCreationMainVM.kt:200)
```

Note: I've tried using the [siteCreationDisabled variable](https://github.com/wordpress-mobile/WordPress-Android/blob/402eb0ae49f3df53741154b41727f8d292975462/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVM.kt#L124) but this didn't work for me probably due to a race condition.

## To reproduce
1. Set a breakpoint at [ImageManager.preload function](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt#L309)
2. Debug a clean version of the WordPress app (from `trunk`)
3. Login
4. Tap the `Create a new site` button
5. Notice that [ImageManager.preload function](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt#L309) is called

## To test
1. Set a breakpoint at [ImageManager.preload function](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt#L309)
2. Debug a clean version of the WordPress app from this branch
3. Login
4. Tap the `Create a new site` button
5. Notice that [ImageManager.preload function](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt#L309) not is called

## Regression Notes
1. Potential unintended areas of impact
N/A

7. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

8. What automated tests I added (or what prevented me from doing so)
Relied on existing site creation tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

antonis committed:
Disables image preloading when site creation is disabled

mkevins commented:
I've tested this, and I noticed two things:

1. On `trunk`, I'm getting the `GlideException` on the login screen (before even tapping on the site creation flow, and before even being logged in). 🤔 
2. I also get this same exception on this branch

The exception I observed also had `onLoadFailed while loading Gravatar image!` in the error message.

I was unable to reproduce the exception (on neither `trunk`, nor this PR) via the site creation flow ("Try the new Jetpack app"). Is it possible there is something subtly different about our steps / environment that could cause a discrepancy?

I tried with various permutations of being logged in and out of Jetpack (and also with Jetpack vanilla installed / not installed), but nothing I tried reproduced the exception described above.

antonis committed:
Prevents attempting to load an empty Gravatar URL

antonis commented:
Thank you for testing and your feedback @mkevins 🙇 

> The exception I observed also had onLoadFailed while loading Gravatar image! in the error message.

Good catch. I think this error log is not fatal and comes from an [attempt to load an empty Gravatar url here](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt#L147). I've pushed a fix to make this more informative with 5fc96d466755fc5c5db1da50cb89cdc26aafb2dc

> I was unable to reproduce the exception (on neither trunk, nor this PR) via the site creation flow ("Try the new Jetpack app"). Is it possible there is something subtly different about our steps / environment that could cause a discrepancy?

I'll try to figure out more detailed steps for this since I have trouble reproducing it consistently myself today 😅

antonis commented:
Hey @mkevins 👋 
I wasn't able to figure out consistent steps to reproduce this locally. I suspect that this might also be triggered by server side reachability since it's a `NoConnectionError`.  

There are currently 9 `GlideException` crashes recorded in Sentry (6 for the Jetpack app and 3 for the WordPress app). Of those the most common one with 4.3k crashes in the past 90 days is [the WordPress crash](https://a8c.sentry.io/share/issue/556f598b913648d39b199ab5c3188bae/) rooting from the [ImageManager.preload function](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/image/ImageManager.kt#L309) that is only used in [preloadThumbnails during site creation](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVM.kt#L202). Given that WordPress.com site creation has been removed from the WordPress app I've added an extra check to prevent this with df402b57fd55a8862439b4e999368d842ae6bcbe 

I updated the testing steps above to be more debug oriented.

AjeshRPai commented:
Hey @antonis and @mkevins 

Not sure if it will help, but I was able to reproduce this issue once by setting the app user language to a non-english one an then trying to create a site.  I suspect this increases the theme response time, but I am not sure. 

I tried to do the same with this PR and couldn't reproduce it. I plan to do a review tomorrow to see if I can reproduce the crash in the trunk and if this PR fixes it. Since this is one of the most common crashes, please feel free to merge this PR once if you think it's enough.

AjeshRPai committed:
↑ Updates: the redundant null check with empty check

AjeshRPai commented on WordPress/src/main/java/org/wordpress/android/ui/main/WPMainNavigationView.kt:
```diff
@@ -134,6 +134,10 @@ class WPMainNavigationView @JvmOverloads constructor(
     }
 
     private fun loadGravatar(imgIcon: ImageView, avatarUrl: String) {
+        if (avatarUrl.isNullOrEmpty()) {
```
Hey @antonis 
There is a null check in the calling place, so updating this logic to check if its empty.  Commit - 7707b6e55f8b4faf2cdc401c6bb067d7d3b0308a

antonis replied:
Thank you for the fix @AjeshRPai 🙇 
This makes sense 👍

AjeshRPai reviewed (approved):
Hey @antonis 
I have tested the following scenarios
1. Preloading thumbnails is not called in Wordpress if the site creation phase is disabled. 🟢 
2. On Jetpack app, the preloading works as expected 🟢 

I made a commit which updates the empty logic for avatar url, commit - 7707b6e55f8b4faf2cdc401c6bb067d7d3b0308a. Hope thats alright. 

I am approving but not merging so that @mkevins  can have a look. But if my review is enough, please go ahead and merge this PR at your convenience. 

Note: I tried to replicate the preload thumbnail crash but couldn't

mkevins reviewed (approved):
I tested via the debugging steps in the description, and this is working as described. Thanks for the fix! Also, thank you Ajesh for the improvement commit.

antonis commented:
Thank you both for reviewing this and the suggestions 🙇
